design/baremetal-operator/uefi-http-boot.md

```diff
@@ -0,0 +1,102 @@
+# Add UEFI HTTP boot hardware driver
+
+## Status
+
+Implementable
+
+## Summary
+
+This design proposes to add a new hardware driver `redfish-http-boot` to
+support the UEFI HTTP boot interface using the Redfish API.
+
+## Motivation
+
+HTTP boot is available since at least 9 years, it was standardized in UEFI
+version 2.5, and it's considered as a natural valid replacement for
+PXE boot.
+It addresses PXE issues for security, relying on HTTPs and using a
+reliable TCP connection, and scalability and performance, for example with
+the possibility of using HTTP load balancers.
+In general HTTP boot grants a faster, more reliable and more secure way to
+obtain operating system images and EFI executables during a network boot
+process.
+
+### Goals
+
+Support UEFI HTTP boot interface using Redfish API.
+
+### Non-Goals
+
+- HTTP boot is a UEFI only feature, so support for Legacy (BIOS) can't and
+won't be taken into consideration.
+- Any non-Redfish support for HTTP boot.
+
+## Proposal
+
+Add a new hardware driver called `redfish-http-boot`, similar to
+`redfish-virtualmedia`, to the Baremetal Operator.
+The new hardware driver supports UEFI HTTP boot using the ironic
+`redfish-https` boot interface.
+
+## Design Details
+
+Using exclusively Redfish API, the accepted value of `Firmware Interface` and
+`RAIDInterface` for the `redfish-http-boot` hardware driver will be `redfish`.  
+Secure Boot will also be supported.
+
+When the `redfish-http-boot` hardware driver is used we need to configure the
+`redfish-https` boot interface in ironic, which is currently not enabled
+in ironic-image.
+
+### Implementation Details/Notes/Constraints
+
+It should be enough to add the new `redfish-http-boot` hardware driver to
+BMO to support the UEFI HTTP boot feature.
+
+### Risks and Mitigations
+
+The UEFI HTTP boot provides a high level of security for a network boot
+process, relying on secure communication between BMC and HTTP server to
+transfer all the files and images needed to boot the host.
+
+### Work Items
+
+- Add a new hardware driver called `redfish-http-boot` to BMO
+- Add `redfish-https` to supported boot interfaces to the ironic-image
+- Update metal3 documentation with details on the new hardware driver and
+how to use it
+
+### Dependencies
+
+Ironic implementation of the `redfish-https` boot interface [has been
+already completed](https://review.opendev.org/c/openstack/ironic/+/900964).
+
+### Test Plan
+
+The new hardwre driver can currently be tested only on supported hardware
+as the Redfish support must be enabled and supported by the BMC.
+
+Unit tests will be implemented as usual.
+
+### Upgrade / Downgrade Strategy
+
+The new hardware driver will be available upgrading the version of Baremetal
+Operator.
+
+### Version Skew Strategy
+
+None
+
+## Drawbacks
+
+None
+
+## Alternatives
+
+The users will keep using other hardware drivers already implemented in
+Baremetal Operator API.
```

Comment:
This section should explain why the existing drivers don't cover all cases, for instance:

1. iPXE is incompatible with secure boot and may have issues on certain platforms.
2. Virtual media is a heavy thing to implement on the BMC side and requires providing the BMC with access to the control plane.